dot/state/service.go

```diff
@@ -44,25 +45,38 @@ func NewService(path string) *Service {
 	return &Service{
 		dbPath:  path,
 		db:      nil,
+		isMemDB: false,
 		Storage: nil,
 		Block:   nil,
 		Network: nil,
 	}
 }
 
+// UseMemDB tells the service to use an in-memory key-value store instead of a persistent database.
+// This should be called after NewService, and before Initialize.
+func (s *Service) UseMemDB() {
+	s.isMemDB = true
+}
+
 // Initialize initializes the genesis state of the DB using the given storage trie. The trie should be loaded with the genesis storage state.
 // The trie does not need a backing DB, since the DB will be created during Service.Start().
 // This only needs to be called during genesis initialization of the node; it doesn't need to be called during normal startup.
 func (s *Service) Initialize(genesisHeader *types.Header, t *trie.Trie) error {
-	datadir, err := filepath.Abs(s.dbPath)
-	if err != nil {
-		return err
-	}
-
-	// initialize database
-	db, err := database.NewBadgerDB(datadir)
-	if err != nil {
-		return err
+	var db database.Database
+	if s.isMemDB {
```

Comment:
is this being used for testing purposes only at the moment? should we make note in a comment here?